src/main/kotlin/uk/gov/justice/hmpps/offenderevents/service/OffenderEventStore.kt

```diff
@@ -6,37 +6,32 @@ import org.springframework.stereotype.Service
 data class StoredMessage(val eventType: EventType, val message: Message)
 
 @Service
-class OffenderEventStore(@Value("\${ui.pageSize}") val pageSize: Int) {
+class OffenderEventStore(@Value("\${model.cacheSize}") val cacheSize: Int,
+                         private val store: MutableList<StoredMessage> = mutableListOf()) : MutableList<StoredMessage> by store {
 
-  private val store: MutableList<StoredMessage> = mutableListOf()
+  override fun add(element: StoredMessage) =
+      store.add(element)
+          .also { if (store.size > cacheSize) store.removeAt(0) }
 
-  fun handleMessage(eventType: EventType, message: Message) {
-    store.add(StoredMessage(eventType, message))
-  }
-
-  fun getAllMessages(eventTypeFilter: String?, textFilter: String?): List<StoredMessage> {
-
-    return store.filterIfNotEmpty(eventTypeFilter) { it.eventType.Value == eventTypeFilter }
-        .filterIfNotEmpty(textFilter) { it.message.Message.contains(textFilter!!.trim()) } // The function does the null check, but the compiler doesn't realise hence bang bang
-        .takeLast(pageSize)
-        .reversed()
-        .toList()
-  }
+  fun handleMessage(message: Message) = add(StoredMessage(message.MessageAttributes.eventType, message))
 
-  fun getAllEventTypes(): List<String> {
-    return store.map { it.eventType.Value }
-        .distinct()
-        .toList()
-  }
+  fun getPageOfMessages(eventTypeFilter: String?, textFilter: String?, pageSize: Int): List<StoredMessage> =
```

Comment:
Improve performance by using a sequence - will not need to traverse entire cache if not applicable